src/commands/commit/prompts.js

```diff
@@ -23,40 +24,51 @@ export type Answers = {
   message: string
 }
 
-export default (gitmojis: Array<Gitmoji>): Array<Object> => [
-  {
-    name: 'gitmoji',
-    message: 'Choose a gitmoji:',
-    type: 'autocomplete',
-    source: (answersSoFor: any, input: string) => {
-      return Promise.resolve(
-        filterGitmojis(input, gitmojis).map((gitmoji) => ({
-          name: `${gitmoji.emoji}  - ${gitmoji.description}`,
-          value: gitmoji[configurationVault.getEmojiFormat()]
-        }))
-      )
+export default (
+  gitmojis: Array<Gitmoji>,
+  mode: 'client' | 'hook'
+): Array<Object> => {
+  const { title, message } =
+    mode === 'hook' ? getDefaultCommitContent() : { title: '', message: '' }
+  return [
+    {
+      name: 'gitmoji',
+      message: 'Choose a gitmoji:',
+      type: 'autocomplete',
+      source: (answersSoFor: any, input: string) => {
+        return Promise.resolve(
+          filterGitmojis(input, gitmojis).map((gitmoji) => ({
+            name: `${gitmoji.emoji}  - ${gitmoji.description}`,
+            value: gitmoji[configurationVault.getEmojiFormat()]
+          }))
+        )
+      }
+    },
+    ...(configurationVault.getScopePrompt()
+      ? [
+          {
+            name: 'scope',
+            message: 'Enter the scope of current changes:',
+            validate: guard.scope
+          }
+        ]
+      : []),
+    {
+      name: 'title',
+      message: 'Enter the commit title',
+      validate: guard.title,
+      transformer: (input: string) => {
+        return `[${
+          (title || input).length
+        }/${TITLE_MAX_LENGTH_COUNT}]: ${input}`
+      },
+      ...(title ? { default: title } : {})
+    },
+    {
+      name: 'message',
+      message: 'Enter the commit message:',
+      validate: guard.message,
+      ...(message ? { default: message } : {})
     }
-  },
-  ...(configurationVault.getScopePrompt()
-    ? [
-        {
-          name: 'scope',
-          message: 'Enter the scope of current changes:',
-          validate: guard.scope
-        }
-      ]
-    : []),
-  {
-    name: 'title',
-    message: 'Enter the commit title',
-    validate: guard.title,
-    transformer: (input: string) => {
-      return `[${input.length}/${TITLE_MAX_LENGTH_COUNT}]: ${input}`
-    }
-  },
-  {
-    name: 'message',
-    message: 'Enter the commit message:',
-    validate: guard.message
-  }
-]
+  ]
+}
```

Comment:
Things we can improve here:

- The `client` | `mode` type should be extracted into the `index` file and imported here to not repeat over and over again

```
// @flow
import inquirer from 'inquirer'

import configurationVault from '../../utils/configurationVault'
import filterGitmojis from '../../utils/filterGitmojis'
import getDefaultCommitContent from '../../utils/getDefaultCommitContent'
import { type CommitMode } from './index'
import guard from './guard'

const TITLE_MAX_LENGTH_COUNT: number = 48

inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'))

export type Gitmoji = {
  code: string,
  description: string,
  emoji: string,
  name: string
}

export type Answers = {
  gitmoji: string,
  scope?: string,
  title: string,
  message: string
}

export default (gitmojis: Array<Gitmoji>, mode: CommitMode): Array<Object> => [
  {
    name: 'gitmoji',
    message: 'Choose a gitmoji:',
    type: 'autocomplete',
    source: (answersSoFor: any, input: string) => {
      return Promise.resolve(
        filterGitmojis(input, gitmojis).map((gitmoji) => ({
          name: `${gitmoji.emoji}  - ${gitmoji.description}`,
          value: gitmoji[configurationVault.getEmojiFormat()]
        }))
      )
    }
  },
  ...(configurationVault.getScopePrompt()
    ? [
        {
          name: 'scope',
          message: 'Enter the scope of current changes:',
          validate: guard.scope
        }
      ]
    : []),
  {
    name: 'title',
    message: 'Enter the commit title',
    validate: guard.title,
    transformer: (input: string) => {
      return `[${
        (title || input).length
      }/${TITLE_MAX_LENGTH_COUNT}]: ${input}`
    },
    default: mode === 'hook'
      ? getDefaultCommitContent().title
      : undefined
  },
  {
    name: 'message',
    message: 'Enter the commit message:',
    validate: guard.message,
    default: mode === 'hook'
      ? getDefaultCommitContent().message
      : undefined
  }
]
```

Reply:
Ok for the type !
For the `default` properties definition on the other hand, I will have to keep the syntax where I initialize `title` and `message` before returning the array.
Indeed, with a syntax like the one you suggested, when running a simple `git commit` without messages, the user will be shown the following, resulting of `default: undefined`
![image](https://user-images.githubusercontent.com/19605940/93653616-f6cd5680-fa19-11ea-814b-3f33461335d6.png)

Moreover, with this syntax, the `getDefaultCommitContent` will be called only once

Reply:
Yes! Makes sense to keep then the variables on the function 😊